cpp/cmake/thirdparty/get_arrow.cmake

```diff
@@ -69,8 +135,13 @@ function(find_and_configure_arrow VERSION BUILD_STATIC ENABLE_S3 ENABLE_ORC ENAB
     set(ARROW_BUILD_STATIC ON)
     set(ARROW_BUILD_SHARED OFF)
     # Turn off CPM using `find_package` so we always download and make sure we get proper static
-    # library
+    # library TODO: Can we use `CPM_DOWNLOAD_<PackageName>` (aka `CPM_DOWNLOAD_ARROW`) instead? This
+    # is a big hammer, although I guess it is effectively limited which packages will be affected by
+    # the scope of the variable here.
```

Comment:
@robertmaynard thoughts on this?

Reply:
Yes we should use `CPM_DOWNLOAD_Arrow` to limit the download